### PR TITLE
UWC : Changes for rebase

### PR DESCRIPTION
**Pull request by aditisa1**
Rebased UWC to EII 3.0

Signed-off-by: aditisa1 <aditi.saxena@intel.com>

### CODE REVIEW

File: eii_configs/Telegraf.json [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
     "config": {
+        "cert_type": ["zmq", "pem"],
```
Comment (nagdeepgk) [inline eii_configs/Telegraf.json:3]: Why are we Generating both zmq and pem for these 2 services - Telegraf and influxdbconnector..

File: kpi-tactic/docker-compose.yml [hunk 2]
```diff
@@ -70,17 +69,13 @@ services:
...
+      - "${EII_INSTALL_PATH}/sockets:${SOCKET_DIR}:z"
       - "${EII_INSTALL_PATH}/uwc_data:/opt/intel/eii/uwc_data:ro"
       - "${EII_INSTALL_PATH}/uwc_data/common_config:${EII_INSTALL_PATH}/uwc_data/common_config:ro"
       - "${EII_INSTALL_PATH}/container_logs/kpi-tactic:/opt/intel/app/logs"
-      secrets:
-        - ca_etcd
-        - etcd_KPIAPP_cert
-        - etcd_KPIAPP_key
-        - ca_broker
-        - client_cert
```
Comment (nagdeepgk) [inline kpi-tactic/docker-compose.yml:82]: this was pointing to mymqttcerts. SO, we need to vol mount even this as we removed the secrets.

File: modbus-master/Dockerfile [hunk 3]
```diff
@@ -30,6 +30,9 @@ WORKDIR /
 
 # compile modbus stack
 COPY modconn /modconn/.
+COPY --from=uwc_common ${CMAKE_INSTALL_PREFIX}/include /usr/local/include
+COPY --from=uwc_common ${CMAKE_INSTALL_PREFIX}/lib /usr/local/lib
```
Comment (nagdeepgk) [inline modbus-master/Dockerfile:34]: this is already mentioned in line 57, 58. Seems a repeat.

File: modbus-master/Dockerfile_RTU [hunk 4]
```diff
@@ -30,6 +30,8 @@ WORKDIR /
 
 COPY modconn /modconn/.
 RUN sed -i 's/-DMODBUS_STACK_TCPIP_ENABLED//g' /modconn/Release/Src/subdir.mk 
+COPY --from=uwc_common ${CMAKE_INSTALL_PREFIX}/include /usr/local/include
```
Comment (nagdeepgk) [inline modbus-master/Dockerfile_RTU:33]: same comments as modbus-TCP

File: modbus-master/modbus_RTU/docker-compose.yml [hunk 5]
```diff
@@ -69,18 +68,16 @@ services:
         max-size: "10m"
     volumes:
     - "${EII_INSTALL_PATH}/uwc_data:/opt/intel/eii/uwc_data:ro"
-    - "vol_uwc_socket:${SOCKET_DIR}:z"
+    - "${EII_INSTALL_PATH}/sockets:${SOCKET_DIR}:z"
```
Comment (nagdeepgk) [inline modbus-master/modbus_RTU/docker-compose.yml:71]: why was this needed.

**Comment by nagdeepgk** [comment]
@vkb1 @Kevin-Midkiff  @iamvishwas  These are the changes done for rebasing UWC to EII-3.0-RC1. Requesting your inputs/review feedback.

File: build_scripts/03_Build_Run_UWC.sh [hunk 6]
```diff
@@ -124,13 +132,37 @@ function harden()
 	docker container update --pids-limit=100 --restart=on-failure:5 --cpu-shares 512 -m 1G --memory-swap -1 modbus-tcp-master
 	docker container update --pids-limit=100 --restart=on-failure:5 --cpu-shares 512 -m 1G --memory-swap -1 mqtt_container
 	docker container update --pids-limit=100 --restart=on-failure:5 --cpu-shares 512 -m 1G --memory-swap -1 modbus-rtu-master
-	docker container update --pids-limit=100 --restart=on-failure:5 --cpu-shares 512 -m 1G --memory-swap -1 ia_etcd
-	docker container update --pids-limit=100 --restart=on-failure:5 --cpu-shares 512 -m 1G --memory-swap -1 ia_etcd_provision
+	#docker container update --pids-limit=100 --restart=on-failure:5 --cpu-shares 512 -m 1G --memory-swap -1 ia_etcd
+	#docker container update --pids-limit=100 --restart=on-failure:5 --cpu-shares 512 -m 1G --memory-swap -1 ia_etcd_provision
```
Comment (vkb1) [inline build_scripts/03_Build_Run_UWC.sh:136]: You may want to have the above 2 lines replaced with ia_configmgr_agent

File: build_scripts/03_Build_Run_UWC.sh [hunk 7]
```diff
@@ -124,13 +132,37 @@ function harden()
...
 	docker ps -q --filter "name=sparkplug-bridge" | grep -q . && docker container update --pids-limit=100 --restart=on-failure:5 --cpu-shares 512 -m 1G --memory-swap -1 sparkplug-bridge
 	# Increase pid limit for KPI to 500 for processing larger count of threads.
 	docker ps -q --filter "name=kpi-tactic-app" | grep -q . && docker container update --pids-limit=500 --restart=on-failure:5 --cpu-shares 512 -m 1G --memory-swap -1 kpi-tactic-app
 }
+mqtt_certs()
+{
+    echo "${GREEN} Genrating certs for mqtt${NC}"	
+    mkdir ./temp ./temp/client ./temp/server
```
Comment (vkb1) [inline build_scripts/03_Build_Run_UWC.sh:144]: You can just use `mkdir -p ./temp/client ./temp/server` which will create temp if it doesn't exist

File: build_scripts/03_Build_Run_UWC.sh [hunk 8]
```diff
@@ -124,13 +132,37 @@ function harden()
...
+    
+    openssl req -config ../Others/mqtt_certs/openssl.cnf -new -newkey rsa:3072 -keyout  ./temp/client/key.pem -out ./temp/client/req.pem -days 3650 -outform PEM -subj /CN=mymqttcerts/O=client/L=$$$/ -nodes
+
+    openssl req -config ../Others/mqtt_certs/openssl.cnf -new -newkey rsa:3072 -keyout  ./temp/server/key.pem -out ./temp/server/req.pem -days 3650 -outform PEM -subj /CN=mymqttcerts/O=server/L=$$$/ -nodes
```
Comment (vkb1) [inline build_scripts/03_Build_Run_UWC.sh:148]: You may want to have vars defined for the openssl.cnf and ./temp/client and ./temp/server as you are using at quite few places heres\

File: build_scripts/03_Build_Run_UWC.sh [hunk 9]
```diff
@@ -124,13 +132,37 @@ function harden()
...
+    openssl ca -days 3650 -cert ../../build/Certificates/rootca/cacert.pem -keyfile ../../build/Certificates/rootca/cakey.pem -in ./temp/client/req.pem -out ./temp/mymqttcerts_client_certificate.pem -outdir ../../build/Certificates/rootca/certs -notext -batch -extensions client_extensions -config ../Others/mqtt_certs/openssl.cnf
+    if [ -e ../../build/Certificates/mymqttcerts ];then
+        rm -r ../../build/Certificates/mymqttcerts
+        
+    fi
+    mkdir ../../build/Certificates/mymqttcerts
+    cp -rf ./temp/mymqttcerts_server_certificate.pem ../../build/Certificates/mymqttcerts/mymqttcerts_server_certificate.pem
+    cp -rf ./temp/mymqttcerts_client_certificate.pem ../../build/Certificates/mymqttcerts/mymqttcerts_client_certificate.pem
+    cp -rf ./temp/server/key.pem  ../../build/Certificates/mymqttcerts/mymqttcerts_server_key.pem
+    cp -rf ./temp/client/key.pem ../../build/Certificates/mymqttcerts/mymqttcerts_client_key.pem
```
Comment (vkb1) [inline build_scripts/03_Build_Run_UWC.sh:161]: Same as above, defining the vars for the prefix path wherever applicable would help

File: modbus-master/Dockerfile [hunk 10]
```diff
@@ -30,6 +30,9 @@ WORKDIR /
...
+RUN ls -a /usr/local/include
```
Comment (vkb1) [inline modbus-master/Dockerfile:35]: This can be removed

File: sparkplug-bridge/docker-compose.yml [hunk 11]
```diff
@@ -71,13 +70,9 @@ services:
...
      - "${EII_INSTALL_PATH}/uwc_data/common_config:${EII_INSTALL_PATH}/uwc_data/common_config:ro"
      - "${EII_INSTALL_PATH}/container_logs/sparkplug-bridge:/opt/intel/app/logs"
-    secrets:
-      - ca_etcd
-      - etcd_SPARKPLUG-BRIDGE_cert
-      - etcd_SPARKPLUG-BRIDGE_key
-      - ca_broker
-      - client_cert
-      - client_key      
+     - ./Certificates/SPARKPLUG-BRIDGE:/run/secrets/SPARKPLUG-BRIDGE
```
Comment (vkb1) [inline sparkplug-bridge/docker-compose.yml:73]: You may want to define small case for naming bind mounts? Mixing CAPS and SMALL use cases may lead to confusion later. 
Also, you can stick to camel case or snake case for consistency.

File: build_scripts/03_Build_Run_UWC.sh (hunk 8 above)
Comment (nagdeepgk) [reply build_scripts/03_Build_Run_UWC.sh:148]: +1

File: build_scripts/03_Build_Run_UWC.sh (hunk 7 above)
Comment (nagdeepgk) [reply build_scripts/03_Build_Run_UWC.sh:144]: +1

File: build_scripts/03_Build_Run_UWC.sh (hunk 9 above)
Comment (nagdeepgk) [reply build_scripts/03_Build_Run_UWC.sh:161]: +1. This would help avoid typos too.

File: modbus-master/Dockerfile (hunk 3 above)
Comment (aditisa1) [reply modbus-master/Dockerfile:34]: i noticed that modconn make is failing as it is unable to locate some libs and include files so added above lines

File: modbus-master/modbus_RTU/docker-compose.yml (hunk 5 above)
Comment (aditisa1) [reply modbus-master/modbus_RTU/docker-compose.yml:71]: Did as per EII

File: build_scripts/02_provision_UWC.sh [hunk 12]
```diff
@@ -616,12 +567,12 @@ if [[ "${IS_SCADA}" -eq "1" ]]; then
...
-    mqtt_certs
+    echo "deployMode="prod"">> ./setenv
 fi
 if [[ "${IS_SCADA}" -eq "1" ]]; then 
    cd  ${Current_Dir}
-  ./2.2_CopyScadaCertsToProvision.sh
+   ./2.2_CopyScadaCertsToProvision.sh
 fi
-harden
+docker rm $(docker ps -qa) --force
```
[the remark this answers is not in the document]
Comment (aditisa1) [reply build_scripts/02_provision_UWC.sh]: added it just for debugging removed it from main code

File: sparkplug-bridge/docker-compose.yml (hunk 11 above)
Comment (aditisa1) [reply sparkplug-bridge/docker-compose.yml:73]: @nagdeepgk what do you suggest 
may have to make changes in many places

Comment (vkb1) [reply sparkplug-bridge/docker-compose.yml:73]: You can have a backlog story and do it later.

**Review by vkb1 — APPROVED** [review]

File: sparkplug-bridge/docker-compose.yml (hunk 11 above)
Comment (nagdeepgk) [reply sparkplug-bridge/docker-compose.yml:73]: yes, can have a backlog story and we can take it up later post RC1. Please add a story @aditisa1 in improvements epic.

**Review by nagdeepgk — APPROVED** [review]

**Comment by nagdeepgk** [comment]
@aditisa1 Need to have readme updates accordingly. I'm merging it now to release tag for val. However, lets do this readme update and create another PR.
